doc/source/ray-references/glossary.rst

```diff
@@ -0,0 +1,306 @@
+Ray Glossary
```

Comment:
Do we consider grouping them based on libraries. The same name may mean different things in different libraries. Maybe using tabs to separate them?

Reply:
My concern with grouping them by libraries is that users, especially new users, don't know different libraries well and it's hard for them to navigate this way. 

Could we start with a flat list grouped by the first character like this https://developers.google.com/machine-learning/glossary#a ? We could change it based on feedback.